Added stellaris ethernet NIC model in Kconfig file.

fadhel086 opened the pull request:
Added CONFIG_NET_QEMU_ETHERNET in overlay file.
This helps to integrate stellaris ethernet with host TAP interface for testing and debugging.

Signed-off-by: Fadhel <fadhel@zilogic.com>

fadhel086 committed:
drivers: ethernet: make run support for stellaris ethernet samples

Added stellaris ethernet NIC model in Kconfig file.
Added CONFIG_NET_QEMU_ETHERNET in overlay file.
This helps to integrate stellaris ethernet with host TAP interface for testing and debugging.

Signed-off-by: Fadhel <fadhel@zilogic.com>

bravegnu commented:
The first line in the commit message should indicate the subsystem:
`drivers: ethernet: ...`

The first line should describe the feature being fixed / added:
`drivers: ethernet: make run support for stellaris ethernet samples`

Description about the change can be added in subsequent lines in the commit message.

fadhel086 committed:
drivers:ethernet: make run support for stellaris ethernet samples(echo-client)

This enables to use net/echo-client with TAP interface in host running a server.
Added overlay-qemu_cortex_m3_eth.conf file.
The app could be build and run using
cmake .. -DBOARD=qemu_cortex_m3 _DOVERLAY_CONFIG=overlay-qemu_cortex_m3_eth.conf
make run.
Choose appropriate ip address for tap interface as mentioned in overlay-qemu_cortex_m3_eth.conf

Signed-off-by: Fadhel <fadhel@zilogic.com>

fadhel086 commented:
> The first line in the commit message should indicate the subsystem:
> `drivers: ethernet: ...`
> 
> The first line should describe the feature being fixed / added:
> `drivers: ethernet: make run support for stellaris ethernet samples`
> 
> Description about the change can be added in subsequent lines in the commit message.

DONE
